### DISCUSSION TITLE
Request not resolved using oca/rest-framework. Error 404

### DISCUSSION THREAD

**Discussion by flachica** [Tecnativa/doodba#477] [Doodba - The template] [answered]
**Question Summary**

In my subproject I have installed _"oca/rest-framework"_ on its 15.0 branch in combination with _"oca/server-auth"_ on its 15.0 branch. If I make requests from the swagger that incorporates this solution, that is, from inside the Odoo container to inside the Odoo container, the system works correctly. On the other hand, if I make requests with Postman, external to the Docker environment, the system gives me a 404 error. I confirm that the request reaches Odoo because it is seen in the log but it does not resolve the URL

**Research done**

As I understand when reading the origin of [Tecnativa/docker-whitelister](https://github.com/Tecnativa/docker-whitelist#why), the problem should not come from a misconfiguration of the docker network, because Doodba [ already account](https://github.com/Tecnativa/doodba-copier-template/blob/main/devel.yaml.jinja#L22) with the opening of necessary ports and, furthermore, it is an external request and not from Odoo to the WWW

I have the suspicion that I have configured something wrong in my subproject, because if I launch Odoo installed from the sources (without Docker) the system does accept requests. I have confirmed with double validation that I am on the latest commit of each of the repositories and that I am always on the 15.0 branch

I would be surprised if it was due to something related to Proxies and the like, so deactivate those options when instantiating my Doodba template

I put the file of my development environment configuration

```
# Changes here will be overwritten by Copier; NEVER EDIT MANUALLY
_commit: v4.0.0
_src_path: gh:Tecnativa/doodba-copier-template
backup_deletion: false
backup_dst: null
backup_email_from: null
backup_email_to: null
backup_image_version: latest
backup_tz: UTC
cidr_whitelist: null
domains_prod: null
domains_test: null
gitlab_url: null
odoo_dbfilter: ^odoo
odoo_initial_lang: es_ES
odoo_listdb: true
odoo_oci_image: null
odoo_proxy: null
odoo_version: 15.0
paths_without_crawlers:
  - /web
  - /website/info
postgres_dbname: odoo
postgres_username: odoo
postgres_version: 12
project_author: Greenice
project_license: no_license
project_name: Odoo-GreenIce
smtp_canonical_default: null
smtp_canonical_domains: null
smtp_default_from: null
smtp_relay_host: null
smtp_relay_port: 587
smtp_relay_user: null
smtp_relay_version: "10"
```

**Comment by pedrobaeza** [comment, marked as answer]
On devel environments, external access is forbidden by default. Check `internal: True` part of the YAML.

**Reply by flachica** [reply]
Dear Baeza, it is somewhat _embarrassing_ but I have found the key and it was something very basic.

After configuring the network correctly (something necessary that you told me) I have realized that in development two databases are always created, "devel" and another "cache-*". **Obviously**, **if in my request I do not send the name** of any of them in the **header**, Odoo **cannot resolve** the URL, since this depends directly on the configuration and installed modules

In devel.yaml I have added

```
services:  
  *****
  odoo:
    environment:
     *****
     DB_FILTER: "^devel"
```

As always, thanks for your input. I am very satisfied with this project. I feel like I finally have a professional environment to develop Odoo/OCA projects. Soon I am going to reconcile with my unfinished business, Docker.

**Reply by pedrobaeza** [reply]
OK, good to know. Such extra DB exists if you use `resetdb`. I prefer to avoid such usage and keep the standard `devel` DB.
